### PR TITLE
chore: standard errors

### PR DESCRIPTION
**Pull request by tjholm**

### CODE REVIEW

File: api/errors/api_error.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	"github.com/nitrictech/go-sdk/api/errors/codes"
+	"google.golang.org/grpc/status"
+)
+
+type ApiError struct {
+	code  codes.Code
+	msg   string
+	cause error
+}
+
```
Comment (medgar-nitric) [inline api/errors/api_error.go:29]: Could be useful to incorporate these Go error improvements: https://blog.golang.org/go1.13-errors
```
func (e *ApiError) Unwrap() error { return e.cause }
```

File: api/errors/api_error.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+	return fmt.Sprintf("%s: %s", a.code.String(), a.msg)
+}
+
+// FromGrpcError - translates a standard grpc error to a nitric api error
+func FromGrpcError(err error) error {
+	if s, ok := status.FromError(err); ok {
+		return &ApiError{
+			code: codes.Code(s.Code()),
+			msg:  s.Message(),
+		}
```
Comment (medgar-nitric) [inline api/errors/api_error.go:45]: what about capturing the cause:
```
		return &ApiError{
			code: codes.Code(s.Code()),
			msg:  s.Message(),
			cause: err,
		}
```

**Review by medgar-nitric — CHANGES_REQUESTED** [review]

File: api/errors/api_error.go (hunk 2 above)
Comment (tjholm) [reply api/errors/api_error.go:45]: This was a conversion, if we want to wrap the orginal error, we can probably delete this method altogether and just use `NewWithCause` instead.

thoughts?

File: api/errors/api_error.go (hunk 1 above)
Comment (tjholm) [reply api/errors/api_error.go:29]: done

File: api/errors/api_error.go (hunk 2 above)
Comment (medgar-nitric) [reply api/errors/api_error.go:49]: I think so yes.

**Review by medgar-nitric — APPROVED** [review]

**Pull request merged**